src/components/authentication/authentication.service.ts

```diff
@@ -404,12 +404,24 @@ export class AuthenticationService {
       )
       .first();
 
-    // remove all the email tokens
+    // remove all the email tokens and invalidate old tokens
     await this.db
       .query()
       .match([node('emailToken', 'EmailToken', { value: result.email })])
       .delete('emailToken')
       .run();
+
+    await this.db
+      .query()
+      .match([
+        node('emailAddress', 'EmailAddress', { value: result.email }),
+        relation('in', '', 'email', { active: true }),
+        node('user', 'User'),
+        relation('out', 'oldRel', 'token', { active: true }),
+        node('token', 'Token'),
+      ])
+      .setValues({ 'oldRel.active': false })
+      .run();
```

Comment:
We still need to avoid the current session here.

Reply:
Do we have a valid `session.token` while resetting the password?

Reply:
I don't think we have `LoggedInSession` in the `resetPassword` resolver.

Reply:
Ah. You can add an anonymous session arg, which has a token, just `userId` is optional.
```ts
@AnonSession() session: Session
```

Reply:
Done